docs/content/docs/data-structures/counter.md

```diff
@@ -1,5 +1,186 @@
 ---
 title: SharedCounter
-menuPosition: 9
-status: unwritten
+menuPosition: 5
 ---
+
+## Introduction
+
+The `SharedCounter` distributed data structure (DDS) is use to store an integer counter value that can be modified collaboratively.
+The data structure affords incrementing and decrementing the shared value via its `increment` method. Decrements are done by providing a negative value.
+
+The `SharedCounter` is a specialized [optimistic DDS]({{< relref "dds.md#optimistic-data-structures" >}}).
+It operates on communicated _deltas_ (amounts by which the shared value should be incremented or decremented), rather than direct changes to the shared value.
+In this way, it avoids the pitfalls of DDSes with simpler merge strategies, in which one user's edit may clobber another's (see [below](#why-a-specialized-dds)).
+
+Note that the `SharedCounter` only operates on integer values.
+
+### Why a specialized DDS?
+
+You may be asking yourself, why not just store the shared integer value directly in another DDS like a [SharedMap][]?
+Why incur the overhead of another runtime type?
+
+The key to the answer here is that DDSes with simpler merge strategies (like `SharedMap`) take a somewhat brute-force approach to merging concurrent edits.
+For a semantic data type like a counter, this can result in undesirable behavior.
+
+#### SharedMap Example
+
+Let's illustrate the issue with an example.
+
+Here, we will use a `SharedMap` to store our shared integer value.
+For simplicity, it will be stored under a static key: `counter-key`.
+
+Let's say that two users are collaborating on an app with a counter widget.
+We will refer to these users as User A and User B.
+The current value of that widget is 42.
+
+What if User A and User B both simultaneously press the `+` button in their UI to increment the current value by 1?
+Behind the scenes, this increment is implemented by writing the updated value to the map.
+That edit is then sequenced and broadcast to other collaborators.
+
+Here, we would expect that, after both users have processed all incoming edits, the resulting value of the counter would be 44.
+Each user pressed the `+` button once.
+42 + 1 + 1 = 44, right?
+
+In fact, because `SharedMap` employs a _last-write-wins_ merge strategy, if User A and User B make their edits at the same time, one of the two updates will be sequnced after the other, and that value will win.
```

Comment:
This new passage is great. One nit: Could we have a one-sentence definition of "at the same time"? I **think** it means that the second op arrives at the relay service before it can broadcast the first op. But, whatever is the correct definition, I think we should put it here. Our readers will know that it is impossible for the two ops to be literally at the same nanosecond. One of those web socket requests will get processed by a NIC on a cloud server before the other.

Reply:
I took a stab at some improvements. The example scenario I chose specifically doesn't care about which user wins, but I called this out more explicitly. Let me know if that makes things more clear, or if it would be better to call out an explicit ordering anyways.